server/ControlPlane/Buffers/Buffers.cs

```diff
@@ -171,20 +236,62 @@ public static void MapBuffers(this WebApplication app)
 
                 bufferUpdate = bufferUpdate with { Id = id };
 
-                var result = await manager.UpdateBufferTags(bufferUpdate, eTagPrecondition, cancellationToken);
+                var result = await manager.UpdateBuffer(bufferUpdate, eTagPrecondition, cancellationToken);
 
                 return result.Match(
                     updated: updated =>
                     {
                         context.Response.Headers.ETag = updated.Value.ETag;
                         return Results.Ok(updated.Value);
                     },
-                    notFound: _ => Results.NotFound(),
-                    preconditionFailed: _ => Results.StatusCode(StatusCodes.Status412PreconditionFailed));
+                    notFound: _ => Responses.NotFound(),
+                    preconditionFailed: failed => Responses.PreconditionFailed(failed.Reason));
             })
             .WithName("setBufferTags")
             .Accepts<BufferUpdate>("application/json")
             .Produces<Buffer>(StatusCodes.Status200OK)
+            .Produces<ErrorBody>(StatusCodes.Status400BadRequest)
+            .Produces<ErrorBody>(StatusCodes.Status404NotFound)
+            .Produces<ErrorBody>(StatusCodes.Status412PreconditionFailed);
+
+        app.MapDelete("/v1/buffers/{id}", async (BufferManager manager, HttpContext context, string id, CancellationToken cancellationToken) =>
+            {
+                if (!context.ParseAndValidateTtlQueryParameter(out var ttl))
+                {
+                    return Responses.BadRequest("ttl must be a valid, non-negative TimeSpan");
+                }
+
+                var softDeleted = false;
+                if (context.Request.Query.TryGetValue("softDeleted", out var softDeletedQuery))
+                {
+                    if (!bool.TryParse(softDeletedQuery, out softDeleted))
+                    {
+                        return Responses.BadRequest("softDeleted must be true or false");
+                    }
+                }
+
+                var result = await manager.SoftDeleteBufferById(id, ttl, softDeleted, cancellationToken);
```

Comment:
If I do 
```
 b=$(tyger buffer create)
 tyger buffer delete $b
 tyger buffer purge $b
 tyger buffer purge $b
```

The second purge gives: 
```
Error: PreconditionFailed: Buffer is expired
```

I think after the purged buffer should be presented as gone at that point (even if the cleanup hasn't happened yet)